ui/component/txoList/view.jsx

```diff
@@ -0,0 +1,264 @@
+
+type Props = {
+  search: string,
+  history: { action: string, push: string => void, replace: string => void },
+  transactionPage: Array<Transaction>,
+  transactionPageNumber: string,
+  transactionItemCount: number,
+  fetchTransactionPage: any => void,
+};
+
+type Delta = {
+  dkey: string,
+  value: string,
+};
+
+function TxoList(props: Props) {
+  const { search, transactionPage, transactionItemCount, fetchTransactionPage, history } = props;
+
+  // parse urlParams
+  const urlParams = new URLSearchParams(search);
+  const page = urlParams.get(TXO.PAGE) || String(1);
+  const pageSize = urlParams.get(TXO.PAGE_SIZE) || String(TX_LIST.PAGE_SIZE);
+  const type = urlParams.get(TXO.TYPE);
+  const subtype = urlParams.get(TXO.SUB_TYPE);
+  const active = urlParams.get(TXO.ACTIVE) || TXO.ACTIVE;
+
+  const currentUrlParams = {
+    page,
+    pageSize,
+    active,
+    type,
+    subtype,
+  };
+
+  // build new json params
+  const params = {};
+
+  if (currentUrlParams.type) {
+    if (currentUrlParams.type === TXO.SENT) {
+      params[TXO.IS_MY_INPUT] = true;
+      params[TXO.IS_NOT_MY_OUTPUT] = true;
+      if (currentUrlParams.subtype === TXO.TIP) {
+        params[TXO.TX_TYPE] = TXO.SUPPORT;
+      } else if (currentUrlParams.subtype === TXO.PURCHASE) {
+        params[TXO.TX_TYPE] = TXO.PURCHASE;
+      } else if (currentUrlParams.subtype === TXO.PURCHASE) {
+        params[TXO.TX_TYPE] = TXO.OTHER;
+      } else {
+        params[TXO.TX_TYPE] = [TXO.OTHER, TXO.PURCHASE, TXO.SUPPORT];
+      }
+    } else if (currentUrlParams.type === TXO.RECEIVED) {
+      params[TXO.IS_MY_OUTPUT] = true;
+      params[TXO.IS_NOT_MY_INPUT] = true;
+      if (currentUrlParams.subtype === TXO.TIP) {
+        params[TXO.TX_TYPE] = TXO.SUPPORT;
+      } else if (currentUrlParams.subtype === TXO.PURCHASE) {
+        params[TXO.TX_TYPE] = TXO.PURCHASE;
+      } else if (currentUrlParams.subtype === TXO.PURCHASE) {
+        params[TXO.TX_TYPE] = TXO.OTHER;
+      } else {
+        params[TXO.TX_TYPE] = [TXO.OTHER, TXO.PURCHASE, TXO.SUPPORT];
+      }
+    } else if (currentUrlParams.type === TXO.SUPPORT) {
+      params[TXO.TX_TYPE] = TXO.SUPPORT;
+      params[TXO.IS_MY_INPUT] = true;
+      params[TXO.IS_MY_OUTPUT] = true;
+    } else if (currentUrlParams.type === TXO.CHANNEL || currentUrlParams.type === TXO.REPOST) {
+      params[TXO.TX_TYPE] = currentUrlParams.type;
+    } else if (currentUrlParams.type === TXO.PUBLISH) {
+      params[TXO.TX_TYPE] = TXO.STREAM;
+    }
+  }
+
+  if (currentUrlParams.active) {
+    if (currentUrlParams.active === 'spent') {
+      params[TXO.IS_SPENT] = true;
+    } else if (currentUrlParams.active === 'active') {
+      params[TXO.IS_NOT_SPENT] = true;
+    }
+  }
+
+  if (currentUrlParams.page) params[TXO.PAGE] = Number(page);
+  if (currentUrlParams.pageSize) params[TXO.PAGE_SIZE] = Number(pageSize);
+
+  function handleChange(delta: Delta) {
+    const url = updateUrl(delta);
+    history.push(url);
```

Comment:
hmmm... replace or push? Do I need/want to paginate through every filter change? (unsure on this)

Reply:
hm, likely replace, as navigating back to page 20 after deleting a bunch of stuff is dumb.